File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        string ref = "aeiou";
        int mx = 0, cur = 0;

        for (int i = 0; i < k; i++) {
            if (ref.find(s[i]) != string::npos) {
                cur++;
            }
        }
        mx = cur;
        for (int i = k; i < s.size(); i++) {
            if (ref.find(s[i]) != string::npos) {
                cur++;
            }
            if (ref.find(s[i - k]) != string::npos) {
                cur--;
            }
            mx = max(mx, cur);
        }
        
        return mx;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/1567-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        ref = "aeiou"
        
        mx = 0
        cur = 0

        for i in range(k):
            if s[i] in ref:
                cur += 1

        mx = cur

        for i in range(k, len(s)):
            if s[i] in ref:
                cur += 1
            
            if s[i-k] in ref:
                cur -= 1

            mx = max(mx, cur)

        return mx